### ISSUE TITLE
Finish the multicurrency feature

### ISSUE THREAD

**Issue by eldy** [Dolibarr/dolibarr#4952]
Introduce multicurrency feature

**Comment by eldy** [comment]
The currency to define "supplier prices" in another currency (not the main) is still missing.

**Comment by caos30** [comment]
Eldy, i would like to help on this feature. I'm new using Dolibarr, but i'm a senior PHP developer. I really need Dolibarr managing perfectly more than one currency. For me and for my customers. I usually make invoices in mexican pesos (MXN), USA dollars and Euros. So... you can understand me ;)

I've installed the 5.0.4 version and i see that multicurrency is extensively developed around the ERP. But i see that product and services in my catalog cannot manage other currency like the the default currency. And i've read on some Github issues that there are wrong outputs of amounts and currencies on some other places of the system (supplier prices, invoices... ).

I have time these next 10 days to work intensively on it. Please, help. I think that i can do a lot to get this solved ASAP.

Cheers! and excellent work with this project. 
Sergi

**Comment by moataztealab** [comment]
sergi plz come back to us if found a solution , it affects the whole structure (suppliers invoices and orders as well as accounting)  which make things really awful...

would be glad to find some help here...

**Comment by caos30** [comment]
Thanks @moataztealab , unfortunately... i had available time in June... now we are on August and i've to work against clock to finish some unfinished works in progress :(

Anyway, i continue trusting on Dolibarr as a good ERP for some of my customers. Actually i'm deploying an instance of Dolibarr to one of them. I choosed 5.0.4 stable version. So my first question that you may be can answer to me is: where must i work on the issues about multicurrency of products/services? on which branch of Github/Dolibarr?

I would prefer on 5.0 branch, but i know that the usual is to do it on "develop" branch. I cannot change my Dolibarr instance for that customer to 6.X yet, mainly because we're using some mexican modules of third parts for digital invoicing and accountancy !!! And they will not have a 6.x Dolibarr version compatible until more than a year at least.

So, please can you tell me how do you work with your customers regarding versions and debugging? I don't understand very well how to do it. It sounds very strange to me to work on debugging on "develop" branch to be available to stable versions of several months in the future. Do you understand me?

Thanks in advance. And i hope that i will be able to help to the project asap.

**Comment by eldy** [comment]
A lot of fixes were introduced in 5.0.7

**Comment by caos30** [comment]
Great!

**Comment by eldy** [comment]
Fix in 6.0.1 is enough to make priority of this feature request lower.

**Comment by eldy** [comment]
Still missing the multicurrency management on template invoice to close the task.

**Comment by franknag** [comment]
Is it possible to prevent Multi-Currency module to get the company default currency?, I've been searching and I can't get the code. If it's possible some conversions could be made because We have created a company using COP (Colombian Peso) and if we set USD as secondary currency the conversion rate will be 1xCOP = 0.00032USD. 1XCOP and that's a $0 Value, different from 3200xCOP = 1xUSD. Any ideas?
![Screen Shot](https://user-images.githubusercontent.com/49418361/55762856-3a045700-5a2a-11e9-8b00-7c83fed3cd7b.png)
